src/sentry/static/sentry/app/types/index.tsx

```diff
@@ -548,14 +549,16 @@ export type DocumentIntegration = {
 };
 
 export type DateString = Date | string | null;
+export type RelativePeriod = keyof typeof DEFAULT_RELATIVE_PERIODS;
+export type IntervalPeriod = ReturnType<typeof getInterval>;
 
 export type GlobalSelection = {
   projects: number[];
   environments: string[];
   datetime: {
     start: DateString;
     end: DateString;
-    period: string;
+    period: RelativePeriod | string; // TODO(ts): Remove "string" and fix type errors
```

Comment:
We might want to keep this `string`. Custom periods like `4h` is possible when hardcoding it in the URL.